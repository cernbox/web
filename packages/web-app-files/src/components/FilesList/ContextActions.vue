<template>
  <context-action-menu :menu-sections="menuSections" :items="items" :space="space" />
</template>

<script lang="ts">
import ContextActionMenu from 'web-pkg/src/components/ContextActions/ContextActionMenu.vue'

import FileActions from '../../mixins/fileActions'
import CreateQuicklink from '../../mixins/actions/createQuicklink'
import EmptyTrashBin from '../../mixins/actions/emptyTrashBin'
import Paste from '../../mixins/actions/paste'
import ShowActions from '../../mixins/actions/showActions'
import ShowDetails from '../../mixins/actions/showDetails'
import ShowShares from '../../mixins/actions/showShares'
import SetSpaceImage from '../../mixins/spaces/actions/setImage'
import SetSpaceReadme from 'web-pkg/src/mixins/spaces/setReadme'
import SpaceNavigate from 'web-pkg/src/mixins/spaces/navigate'
import { PropType } from 'vue'
import { Resource } from 'web-client'
import { SpaceResource } from 'web-client/src/helpers'
import DeletedFiles from 'web-pkg/src/mixins/spaces/deletedFiles'

export default {
  name: 'ContextActions',
  components: { ContextActionMenu },
  mixins: [
    FileActions,
    DeletedFiles,
    CreateQuicklink,
    EmptyTrashBin,
    Paste,
    ShowActions,
    ShowDetails,
    ShowShares,
    SetSpaceImage,
    SetSpaceReadme,
    SpaceNavigate
  ],

  props: {
    space: {
      type: Object as PropType<SpaceResource>,
      required: true
    },
    items: {
      type: Array as PropType<Resource[]>,
      required: true
    },
    deactivateApps: {
      type: Boolean,
      required: false,
      default: false
    }
  },

  computed: {
    menuSections() {
      const sections = []
      if (this.items.length > 1) {
        if (this.menuItemsBatchActions.length) {
          sections.push({
            name: 'batch-actions',
            items: [...this.menuItemsBatchActions]
          })
        }
        sections.push({
          name: 'batch-details',
          items: [...this.$_showDetails_items]
        })
        return sections
      }

      if (this.menuItemsContext.length && !this.deactivateApps) {
        sections.push({
          name: 'context',
          items: this.menuItemsContext
        })
      }
      if (this.menuItemsShare.length) {
        sections.push({
          name: 'share',
          items: this.menuItemsShare
        })
      }
      if (this.menuItemsActions.length) {
        sections.push({
          name: 'actions',
          items: this.menuItemsActions
        })
      }
      if (this.menuItemsSidebar.length) {
        sections.push({
          name: 'sidebar',
          items: this.menuItemsSidebar
        })
      }
      return sections
    },

    filterParams() {
      return {
        space: this.space,
        resources: this.items
      }
    },

    menuItemsBatchActions() {
      return [
        ...this.$_acceptShare_items,
        ...this.$_declineShare_items,
        ...this.$_downloadArchive_items,
        ...this.$_delete_items,
        ...this.$_move_items,
        ...this.$_copy_items,
        ...this.$_emptyTrashBin_items,
        ...this.$_restore_items
      ].filter((item) => item.isEnabled(this.filterParams))
    },

    menuItemsContext() {
      const fileHandlers = [
        ...this.$_fileActions_editorActions,
        ...this.$_fileActions_loadExternalAppActions(this.filterParams)
      ]

      return [...fileHandlers]
        .filter((item) => item.isEnabled(this.filterParams))
        .sort((x, y) => Number(y.canBeDefault) - Number(x.canBeDefault))
    },

    menuItemsShare() {
      return [...this.$_showShares_items, ...this.$_createQuicklink_items].filter((item) =>
        item.isEnabled(this.filterParams)
      )
    },

    menuItemsActions() {
      return [
        ...this.$_downloadArchive_items,
        ...this.$_downloadFile_items,
        ...this.$_delete_items,
        ...this.$_move_items,
        ...this.$_copy_items,
        ...this.$_paste_items,
        ...this.$_rename_items,
        ...this.$_showEditTags_items,
        ...this.$_restore_items,
        ...this.$_acceptShare_items,
        ...this.$_declineShare_items,
        ...this.$_setSpaceImage_items,
        ...this.$_setSpaceReadme_items,
        ...this.$_navigate_space_items
      ].filter((item) => item.isEnabled(this.filterParams))
    },

    menuItemsSidebar() {
      const fileHandlers = [...this.$_navigate_items]

      return [
        ...this.$_favorite_items.map((action) => {
          action.keepOpen = true
          return action
        }),
        ...fileHandlers,
        ...this.$_deletedFiles_items_generic,
        ...this.$_showDetails_items
      ].filter((item) => item.isEnabled(this.filterParams))
    }
  }
}
</script>
