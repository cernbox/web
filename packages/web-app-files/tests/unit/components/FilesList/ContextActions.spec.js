import Vuex from 'vuex'
import DesignSystem from 'owncloud-design-system'
import stubs from '@/tests/unit/stubs/index.js'
import { createLocalVue, mount } from '@vue/test-utils'
import ContextActions from '../../../../src/components/FilesList/ContextActions.vue'
import GetTextPlugin from 'vue-gettext'
import fixtureMimeTypes from '@files/tests/__fixtures__/mimeTypes.js'

const localVue = createLocalVue()
localVue.use(DesignSystem)
localVue.use(Vuex)
localVue.use(GetTextPlugin, {
  translations: 'does-not-matter.json',
  silent: true
})

const exampleApps = [
  { name: 'exampleName1', icon: 'exampleIcon1' },
  { name: 'exampleName2', icon: 'exampleIcon2' },
  { name: 'exampleName3', icon: 'exampleIcon3' }
]

const mockMenuSections = [
  {
    name: 'context',
    items: [
      {
        icon: 'text',
        canBeDefault: true,
        handler: jest.fn(),
        label: () => 'Open in editor',
        componentType: 'oc-button',
        selector: '.oc-files-actions-markdown-editor-trigger',
        class: 'oc-files-actions-markdown-editor-trigger'
      },
      {
        icon: 'file_download',
        canBeDefault: true,
        handler: jest.fn(),
        label: () => 'Download',
        componentType: 'oc-button',
        selector: '.oc-files-actions-download-file-trigger',
        class: 'oc-files-actions-download-file-trigger'
      },
      {
        icon: 'link-add',
        handler: jest.fn(),
        label: () => 'Create link',
        componentType: 'oc-button',
        selector: '.oc-files-actions-create-public-link-trigger',
        class: 'oc-files-actions-create-public-link-trigger'
      },
      {
        icon: 'group-add',
        handler: jest.fn(),
        label: () => 'Share',
        componentType: 'oc-button',
        selector: '.oc-files-actions-show-shares-trigger',
        class: 'oc-files-actions-show-shares-trigger'
      }
    ]
  },
  {
    name: 'actions',
    items: [
      {
        icon: 'edit',
        handler: jest.fn(),
        label: () => 'Edit',
        componentType: 'oc-button',
        selector: '.oc-files-actions-rename-trigger',
        class: 'oc-files-actions-rename-trigger'
      },
      {
        icon: 'folder-move',
        handler: jest.fn(),
        label: () => 'Move',
        componentType: 'oc-button',
        selector: '.oc-files-actions-move-trigger',
        class: 'oc-files-actions-move-trigger'
      },
      {
        icon: 'file_copy',
        handler: jest.fn(),
        label: () => 'Copy',
        componentType: 'oc-button',
        selector: '.oc-files-actions-copy-trigger',
        class: 'oc-files-actions-copy-trigger'
      },
      {
        icon: 'delete',
        handler: jest.fn(),
        label: () => 'Delete',
        componentType: 'oc-button',
        selector: '.oc-files-actions-delete-trigger',
        class: 'oc-files-actions-delete-trigger'
      },
      {
        icon: 'slideshow',
        handler: jest.fn(),
        label: () => 'All actions',
        componentType: 'oc-button',
        selector: '.oc-files-actions-show-actions-trigger',
        class: 'oc-files-actions-show-actions-trigger'
      }
    ]
  },
  {
    name: 'sidebar',
    items: [
      {
        icon: 'info_outline',
        handler: jest.fn(),
        label: () => 'Details',
        componentType: 'oc-button',
        selector: '.oc-files-actions-show-details-trigger',
        class: 'oc-files-actions-show-details-trigger'
      }
    ]
  }
]

const filesPersonalRoute = { name: 'files-personal' }

describe('ContextActions', () => {
  describe('if no externalApps are available, renders', () => {
    it('a list of actions for a file', () => {
      const wrapper = getWrapper(filesPersonalRoute, {
        name: 'exampleFile',
        extension: 'jpg',
        mimeType: 'application/fileFormat2',
        type: 'file'
      })

      expect(wrapper).toMatchSnapshot()
    })

    it('a list of actions for a folder', () => {
      const wrapper = getWrapper(filesPersonalRoute, {
        name: 'exampleFolder',
        extension: '',
        type: 'folder'
      })

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('if externalApps are available, renders', () => {
    it('two lists of actions for a file', () => {
      const wrapper = getWrapper(
        filesPersonalRoute,
        {
          name: 'exampleFile',
          extension: 'jpg',
          mimeType: 'application/fileFormat2',
          type: 'file'
        },
        exampleApps
      )

      expect(wrapper).toMatchSnapshot()
    })

    it('one list of actions for a folder', () => {
      const wrapper = getWrapper(
        filesPersonalRoute,
        {
          name: 'exampleFolder',
          extension: '',
          type: 'folder'
        },
        exampleApps
      )

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('should trigger the action handlers on click', () => {
    afterEach(() => {
      jest.clearAllMocks()
    })

    it('for external App actions', async () => {
      const triggerActionSpy = jest
        .spyOn(ContextActions.mixins[0].methods, '$_fileActions_openLink')
        .mockImplementationOnce(jest.fn)

      const wrapper = getWrapper(
        filesPersonalRoute,
        {
          name: 'exampleFile',
          extension: 'jpg',
          mimeType: 'application/fileFormat2',
          type: 'file'
        },
        exampleApps
      )

      const app = exampleApps[0]
      const appButton = wrapper.find('#oc-files-context-default-actions > li > button')
      expect(appButton.text()).toBe(`Open in ${app.name}`)

      await appButton.trigger('click')
      expect(triggerActionSpy).toHaveBeenCalledWith(app.name, 'a93f8adf==')
    })

    it('for default file and system actions', async () => {
      const wrapper = getWrapper(
        filesPersonalRoute,
        {
          name: 'exampleFile',
          extension: 'jpg',
          mimeType: 'application/fileFormat2',
          type: 'file'
        },
        exampleApps
      )

      for (const section of mockMenuSections) {
        for (const item of section.items) {
          const buttonElement = wrapper.find(item.selector)
          expect(buttonElement.exists()).toBeTruthy()
          await buttonElement.trigger('click')
          expect(item.handler).toHaveBeenCalledTimes(1)
        }
      }
    })
  })
})

function getWrapper(route, { filename, extension, type = '', mimeType }, availableApps = []) {
  const mountStubs = { ...stubs, 'oc-button': false }

  return mount(ContextActions, {
    localVue,
    store: createStore(),
    data() {
      return {
        appList: availableApps
      }
    },
    stubs: mountStubs,
    mocks: {
      $route: route,
      publicPage: () => false
    },
    propsData: {
      item: {
        id: 'a93f8adf==',
        name: filename,
        path: type === 'file' ? `/${filename}.${extension}` : `/${filename}`,
        mimeType,
        extension,
        type,
        canDownload: () => true,
        isReceivedShare: () => true,
        canBeDeleted: () => true,
        canRename: () => true
      }
    },
    computed: {
      menuSections: () => mockMenuSections
    }
  })
}

function createStore(state) {
  return new Vuex.Store({
    getters: {
      capabilities: jest.fn(() => ({
        files: {
          app_providers: [
            {
              apps_url: '/app/list',
              enabled: true,
              open_url: '/app/open'
            }
          ]
        }
      }))
    },
    modules: {
      External: {
        state: {
          ...state
        },
        namespaced: true,
        getters: {
          getMimeTypes: () => {
            return fixtureMimeTypes
          }
        }
      },
      Files: {
        state: {
          ...state
        },
        namespaced: true,
        getters: {
          currentFolder: () => '/'
        }
      }
    }
  })
}
