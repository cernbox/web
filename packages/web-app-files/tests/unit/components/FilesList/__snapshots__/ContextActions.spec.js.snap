// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContextActions if externalApps are available, renders one list of actions for a folder 1`] = `
<div id="oc-files-context-menu">
  <!---->
  <!---->
  <ul id="\`oc-files-context-actions-\${section.name}\`" class="uk-list oc-mt-s oc-files-context-actions">
    <li class="oc-files-context-action"><button class="oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-text-bold oc-files-actions-markdown-editor-trigger">
        <oc-icon-stub name="text" accessiblelabel="" type="span" size="medium" variation="passive"></oc-icon-stub> <span class="oc-files-context-action-label">Open in editor</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action"><button class="oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-text-bold oc-files-actions-download-file-trigger">
        <oc-icon-stub name="file_download" accessiblelabel="" type="span" size="medium" variation="passive"></oc-icon-stub> <span class="oc-files-context-action-label">Download</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action"><button class="oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-text-bold oc-files-actions-create-public-link-trigger">
        <oc-icon-stub name="link-add" accessiblelabel="" type="span" size="medium" variation="passive"></oc-icon-stub> <span class="oc-files-context-action-label">Create link</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action"><button class="oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-text-bold oc-files-actions-show-shares-trigger">
        <oc-icon-stub name="group-add" accessiblelabel="" type="span" size="medium" variation="passive"></oc-icon-stub> <span class="oc-files-context-action-label">Share</span>
        <!---->
      </button></li>
  </ul>
  <hr>
  <ul id="\`oc-files-context-actions-\${section.name}\`" class="uk-list oc-mt-s oc-files-context-actions">
    <li class="oc-files-context-action"><button class="oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-text-bold oc-files-actions-rename-trigger">
        <oc-icon-stub name="edit" accessiblelabel="" type="span" size="medium" variation="passive"></oc-icon-stub> <span class="oc-files-context-action-label">Edit</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action"><button class="oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-text-bold oc-files-actions-move-trigger">
        <oc-icon-stub name="folder-move" accessiblelabel="" type="span" size="medium" variation="passive"></oc-icon-stub> <span class="oc-files-context-action-label">Move</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action"><button class="oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-text-bold oc-files-actions-copy-trigger">
        <oc-icon-stub name="file_copy" accessiblelabel="" type="span" size="medium" variation="passive"></oc-icon-stub> <span class="oc-files-context-action-label">Copy</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action"><button class="oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-text-bold oc-files-actions-delete-trigger">
        <oc-icon-stub name="delete" accessiblelabel="" type="span" size="medium" variation="passive"></oc-icon-stub> <span class="oc-files-context-action-label">Delete</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action"><button class="oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-text-bold oc-files-actions-show-actions-trigger">
        <oc-icon-stub name="slideshow" accessiblelabel="" type="span" size="medium" variation="passive"></oc-icon-stub> <span class="oc-files-context-action-label">All actions</span>
        <!---->
      </button></li>
  </ul>
  <hr>
  <ul id="\`oc-files-context-actions-\${section.name}\`" class="uk-list oc-mt-s oc-files-context-actions">
    <li class="oc-files-context-action"><button class="oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-text-bold oc-files-actions-show-details-trigger">
        <oc-icon-stub name="info_outline" accessiblelabel="" type="span" size="medium" variation="passive"></oc-icon-stub> <span class="oc-files-context-action-label">Details</span>
        <!---->
      </button></li>
  </ul>
  <!---->
</div>
`;

exports[`ContextActions if externalApps are available, renders two lists of actions for a file 1`] = `
<div id="oc-files-context-menu">
  <ul id="oc-files-context-default-actions" class="uk-list oc-my-xs oc-files-context-actions">
    <li><button class="oc-text-bold oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw"><img src="exampleIcon1" alt="Icon for exampleName1 app" class="oc-icon oc-icon-m"> <span class="oc-files-context-action-label">Open in exampleName1</span></button></li>
    <li><button class="oc-text-bold oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw"><img src="exampleIcon2" alt="Icon for exampleName2 app" class="oc-icon oc-icon-m"> <span class="oc-files-context-action-label">Open in exampleName2</span></button></li>
    <li><button class="oc-text-bold oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw"><img src="exampleIcon3" alt="Icon for exampleName3 app" class="oc-icon oc-icon-m"> <span class="oc-files-context-action-label">Open in exampleName3</span></button></li>
  </ul>
  <hr>
  <ul id="\`oc-files-context-actions-\${section.name}\`" class="uk-list oc-mt-s oc-files-context-actions">
    <li class="oc-files-context-action"><button class="oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-text-bold oc-files-actions-markdown-editor-trigger">
        <oc-icon-stub name="text" accessiblelabel="" type="span" size="medium" variation="passive"></oc-icon-stub> <span class="oc-files-context-action-label">Open in editor</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action"><button class="oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-text-bold oc-files-actions-download-file-trigger">
        <oc-icon-stub name="file_download" accessiblelabel="" type="span" size="medium" variation="passive"></oc-icon-stub> <span class="oc-files-context-action-label">Download</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action"><button class="oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-text-bold oc-files-actions-create-public-link-trigger">
        <oc-icon-stub name="link-add" accessiblelabel="" type="span" size="medium" variation="passive"></oc-icon-stub> <span class="oc-files-context-action-label">Create link</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action"><button class="oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-text-bold oc-files-actions-show-shares-trigger">
        <oc-icon-stub name="group-add" accessiblelabel="" type="span" size="medium" variation="passive"></oc-icon-stub> <span class="oc-files-context-action-label">Share</span>
        <!---->
      </button></li>
  </ul>
  <hr>
  <ul id="\`oc-files-context-actions-\${section.name}\`" class="uk-list oc-mt-s oc-files-context-actions">
    <li class="oc-files-context-action"><button class="oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-text-bold oc-files-actions-rename-trigger">
        <oc-icon-stub name="edit" accessiblelabel="" type="span" size="medium" variation="passive"></oc-icon-stub> <span class="oc-files-context-action-label">Edit</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action"><button class="oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-text-bold oc-files-actions-move-trigger">
        <oc-icon-stub name="folder-move" accessiblelabel="" type="span" size="medium" variation="passive"></oc-icon-stub> <span class="oc-files-context-action-label">Move</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action"><button class="oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-text-bold oc-files-actions-copy-trigger">
        <oc-icon-stub name="file_copy" accessiblelabel="" type="span" size="medium" variation="passive"></oc-icon-stub> <span class="oc-files-context-action-label">Copy</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action"><button class="oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-text-bold oc-files-actions-delete-trigger">
        <oc-icon-stub name="delete" accessiblelabel="" type="span" size="medium" variation="passive"></oc-icon-stub> <span class="oc-files-context-action-label">Delete</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action"><button class="oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-text-bold oc-files-actions-show-actions-trigger">
        <oc-icon-stub name="slideshow" accessiblelabel="" type="span" size="medium" variation="passive"></oc-icon-stub> <span class="oc-files-context-action-label">All actions</span>
        <!---->
      </button></li>
  </ul>
  <hr>
  <ul id="\`oc-files-context-actions-\${section.name}\`" class="uk-list oc-mt-s oc-files-context-actions">
    <li class="oc-files-context-action"><button class="oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-text-bold oc-files-actions-show-details-trigger">
        <oc-icon-stub name="info_outline" accessiblelabel="" type="span" size="medium" variation="passive"></oc-icon-stub> <span class="oc-files-context-action-label">Details</span>
        <!---->
      </button></li>
  </ul>
  <!---->
</div>
`;

exports[`ContextActions if no externalApps are available, renders a list of actions for a file 1`] = `
<div id="oc-files-context-menu">
  <!---->
  <!---->
  <ul id="\`oc-files-context-actions-\${section.name}\`" class="uk-list oc-mt-s oc-files-context-actions">
    <li class="oc-files-context-action"><button class="oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-text-bold oc-files-actions-markdown-editor-trigger">
        <oc-icon-stub name="text" accessiblelabel="" type="span" size="medium" variation="passive"></oc-icon-stub> <span class="oc-files-context-action-label">Open in editor</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action"><button class="oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-text-bold oc-files-actions-download-file-trigger">
        <oc-icon-stub name="file_download" accessiblelabel="" type="span" size="medium" variation="passive"></oc-icon-stub> <span class="oc-files-context-action-label">Download</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action"><button class="oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-text-bold oc-files-actions-create-public-link-trigger">
        <oc-icon-stub name="link-add" accessiblelabel="" type="span" size="medium" variation="passive"></oc-icon-stub> <span class="oc-files-context-action-label">Create link</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action"><button class="oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-text-bold oc-files-actions-show-shares-trigger">
        <oc-icon-stub name="group-add" accessiblelabel="" type="span" size="medium" variation="passive"></oc-icon-stub> <span class="oc-files-context-action-label">Share</span>
        <!---->
      </button></li>
  </ul>
  <hr>
  <ul id="\`oc-files-context-actions-\${section.name}\`" class="uk-list oc-mt-s oc-files-context-actions">
    <li class="oc-files-context-action"><button class="oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-text-bold oc-files-actions-rename-trigger">
        <oc-icon-stub name="edit" accessiblelabel="" type="span" size="medium" variation="passive"></oc-icon-stub> <span class="oc-files-context-action-label">Edit</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action"><button class="oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-text-bold oc-files-actions-move-trigger">
        <oc-icon-stub name="folder-move" accessiblelabel="" type="span" size="medium" variation="passive"></oc-icon-stub> <span class="oc-files-context-action-label">Move</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action"><button class="oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-text-bold oc-files-actions-copy-trigger">
        <oc-icon-stub name="file_copy" accessiblelabel="" type="span" size="medium" variation="passive"></oc-icon-stub> <span class="oc-files-context-action-label">Copy</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action"><button class="oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-text-bold oc-files-actions-delete-trigger">
        <oc-icon-stub name="delete" accessiblelabel="" type="span" size="medium" variation="passive"></oc-icon-stub> <span class="oc-files-context-action-label">Delete</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action"><button class="oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-text-bold oc-files-actions-show-actions-trigger">
        <oc-icon-stub name="slideshow" accessiblelabel="" type="span" size="medium" variation="passive"></oc-icon-stub> <span class="oc-files-context-action-label">All actions</span>
        <!---->
      </button></li>
  </ul>
  <hr>
  <ul id="\`oc-files-context-actions-\${section.name}\`" class="uk-list oc-mt-s oc-files-context-actions">
    <li class="oc-files-context-action"><button class="oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-text-bold oc-files-actions-show-details-trigger">
        <oc-icon-stub name="info_outline" accessiblelabel="" type="span" size="medium" variation="passive"></oc-icon-stub> <span class="oc-files-context-action-label">Details</span>
        <!---->
      </button></li>
  </ul>
  <!---->
</div>
`;

exports[`ContextActions if no externalApps are available, renders a list of actions for a folder 1`] = `
<div id="oc-files-context-menu">
  <!---->
  <!---->
  <ul id="\`oc-files-context-actions-\${section.name}\`" class="uk-list oc-mt-s oc-files-context-actions">
    <li class="oc-files-context-action"><button class="oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-text-bold oc-files-actions-markdown-editor-trigger">
        <oc-icon-stub name="text" accessiblelabel="" type="span" size="medium" variation="passive"></oc-icon-stub> <span class="oc-files-context-action-label">Open in editor</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action"><button class="oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-text-bold oc-files-actions-download-file-trigger">
        <oc-icon-stub name="file_download" accessiblelabel="" type="span" size="medium" variation="passive"></oc-icon-stub> <span class="oc-files-context-action-label">Download</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action"><button class="oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-text-bold oc-files-actions-create-public-link-trigger">
        <oc-icon-stub name="link-add" accessiblelabel="" type="span" size="medium" variation="passive"></oc-icon-stub> <span class="oc-files-context-action-label">Create link</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action"><button class="oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-text-bold oc-files-actions-show-shares-trigger">
        <oc-icon-stub name="group-add" accessiblelabel="" type="span" size="medium" variation="passive"></oc-icon-stub> <span class="oc-files-context-action-label">Share</span>
        <!---->
      </button></li>
  </ul>
  <hr>
  <ul id="\`oc-files-context-actions-\${section.name}\`" class="uk-list oc-mt-s oc-files-context-actions">
    <li class="oc-files-context-action"><button class="oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-text-bold oc-files-actions-rename-trigger">
        <oc-icon-stub name="edit" accessiblelabel="" type="span" size="medium" variation="passive"></oc-icon-stub> <span class="oc-files-context-action-label">Edit</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action"><button class="oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-text-bold oc-files-actions-move-trigger">
        <oc-icon-stub name="folder-move" accessiblelabel="" type="span" size="medium" variation="passive"></oc-icon-stub> <span class="oc-files-context-action-label">Move</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action"><button class="oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-text-bold oc-files-actions-copy-trigger">
        <oc-icon-stub name="file_copy" accessiblelabel="" type="span" size="medium" variation="passive"></oc-icon-stub> <span class="oc-files-context-action-label">Copy</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action"><button class="oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-text-bold oc-files-actions-delete-trigger">
        <oc-icon-stub name="delete" accessiblelabel="" type="span" size="medium" variation="passive"></oc-icon-stub> <span class="oc-files-context-action-label">Delete</span>
        <!---->
      </button></li>
    <li class="oc-files-context-action"><button class="oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-text-bold oc-files-actions-show-actions-trigger">
        <oc-icon-stub name="slideshow" accessiblelabel="" type="span" size="medium" variation="passive"></oc-icon-stub> <span class="oc-files-context-action-label">All actions</span>
        <!---->
      </button></li>
  </ul>
  <hr>
  <ul id="\`oc-files-context-actions-\${section.name}\`" class="uk-list oc-mt-s oc-files-context-actions">
    <li class="oc-files-context-action"><button class="oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-text-bold oc-files-actions-show-details-trigger">
        <oc-icon-stub name="info_outline" accessiblelabel="" type="span" size="medium" variation="passive"></oc-icon-stub> <span class="oc-files-context-action-label">Details</span>
        <!---->
      </button></li>
  </ul>
  <!---->
</div>
`;
